app/services/import/persist/persist-licence-document.service.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+
+/**
+ * Creates or updates a licence document
+ * @module PersistLicenceDocumentService
+ */
+
+const LicenceDocumentModel = require('../../../models/licence-document.model.js')
+
+/**
+ * Creates or updates a licence document
+ *
+ * @param {object} trx - An Objection.js transaction object for PostgreSQL.
+ * @param {string} updatedAt - The timestamp indicating when the entity was last updated.
+ * @param {object} transformedLicence - An object representing a valid WRLS licence.
+ *
+ * @returns {Promise<string>} - The licence ID from WRLS.
+ */
+async function go (trx, updatedAt, transformedLicence) {
+  await _persistLicenceDocument(trx, updatedAt, transformedLicence.licenceDocument)
+}
+
+async function _persistLicenceDocument (trx, updatedAt, licenceDocument) {
+  return LicenceDocumentModel.query(trx)
+    .insert({ ...licenceDocument, updatedAt })
+    .onConflict('licenceRef')
+    .merge([
+      'endDate',
+      'startDate',
+      'updatedAt',
+      'deletedAt'
```

Comment:
I'd have to dig into how `deletedAt` is determined (because I think that is set during the import as well).

But we shouldn't be overwriting this if it is already set.

Reply:
I see what you mean now.

Reply:
```
-- auto-generated definition
create table documents
(
    document_id   uuid      default gen_random_uuid()                        not null
        primary key,
    regime        varchar   default 'water'::character varying               not null,
    document_type varchar   default 'abstraction_licence'::character varying not null,
    document_ref  varchar                                                    not null,
    start_date    date                                                       not null,
    end_date      date,
    external_id   varchar,
    date_created  timestamp default now()                                    not null,
    date_updated  timestamp default now()                                    not null,
    is_test       boolean   default false                                    not null,
    date_deleted  timestamp
);

```

I can remove `'deletedAt'` from the upsert.